docs/deployment/resource-providers/yarn.md

```diff
@@ -0,0 +1,200 @@
+
+### Session Mode
+
+We describe deployment with the Session Mode in the [Getting Started](#getting-started) guide at the top of the page.
+
+The Session Mode has two operation modes:
+- **attached mode** (default): The yarn-session.sh client submits the Flink clsuter to YARN, but the client keeps running, tracking the state of the cluster. If the cluster fails, the client will show the error. If the client gets terminated, it will signal the cluster to shut down as well.
+- **detached mode**: The yarn-session.sh client submits the Flink cluster to YARN, then the client returns. Another invocation of the client, or YARN tools is needed to stop the Flink cluster.
+
+The session mode will create a hidden YARN properties file in `/tmp/.yarn-properties-<username>`, which will be picked up for cluster discovery by the command line interface when submitting a job.
+
+You can re-attach to a YARN session using the following command:
+
+```
+./bin/yarn-session.sh -yid application_XXXX_YY
+```
+
+Besides passing [configuration]({% link deployment/config.md %}) via the `conf/flink-conf.yaml` file, you can also pass any configuration at submission time to the `./bin/yarn-session.sh` client using `-Dkey=value` arguments.
+
+The YARN session client also has a few "shortcut arguments" for commonly used settings. They can be listed with `./bin/yarn-session.sh -h`.
+
+
+## Flink on YARN Reference
+
+### Configuring Flink on YARN
+
+The YARN-specific configurations are listed on the [configuration page]({% link deployment/config.md %}#yarn).
+
+The following configuration parameters have no effect in Flink on YARN, as they are overwritten by the framework at runtime:
+- `jobmanager.rpc.address` (dynamically set to the address of the JobManager container by Flink on YARN)
+- `io.tmp.dirs` (Flink sets the temporary directories defined by YARN)
+
+
+### Resource Allocation Behavior
+
+A JobManager submitted in YARN session mode will wait for job submissions before starting to request and allocate TaskManagers. The other deployment modes are immediately requesting the required TaskManagers from YARN.
+
+The memory configurations for JobManager and TaskManager processes will be respected by the YARN implementation. The number of reported VCores is by default equal to the number of configured slots per TaskManager. The [yarn.containers.vcores]({% link deployment/config.md %}#yarn-containers-vcores) allows overwriting the number of vcores with a custom value. In order for this parameter to work you should enable CPU scheduling in your YARN cluster.
+
+Failed containers (including the JobManager) are replaced by YARN. The maximum number of JobManager container restarts is configured via [yarn.application-attempts]({% link deployment/config.md %}#yarn-application-attempts) (default 1). The YARN Application will fail once all attempts are exhausted.
+
+
+### High-Availability on YARN
```

Comment:
Here is an interesting piece of information from the HA documentation page:

Important: You should not set this value manually when running a YARN cluster, a per-job YARN session, or on another cluster manager. In those cases a cluster-id is automatically being generated based on the application id. Manually setting a cluster-id overrides this behaviour in YARN.

Reply:
Very good point! thx